### ISSUE TITLE
Video Scraping Does Not work for BrandNewTube.com

### ISSUE THREAD

**Issue by 4nanook** [friendica/friendica#10779]
- [x ] I have searched open and closed issues for duplicates

### Bug Description

Video scraping does not work for brandnewtube.com URLs.
### Steps to Reproduce

1. Post into the link or video option a brandnewtube.com URL.

Actual Result:

It just shows the URL as a link, no preview.

Expected Result:

A video preview as you get with youtube, bitchute, odysee, etc.

### Platform Info

Friendica Version:

Friendica Source: 
Github
PHP version:
PHP 7.4.24 (cli) (built: Sep 23 2021 21:36:11) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.24, Copyright (c), by Zend Technologies

SQL version:
mysql  Ver 15.1 Distrib 10.5.12-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

**Comment by MrPetovan** [comment]
I don't think brandnewtube.com is general purpose enough that we should enable video embedding of their URLs in Friendica.

**Comment by annando** [comment]
The data we present in a link preview is the data that the remote side does offer. When we don't display any data than it is most likely that they don't offer anything that can be used. This is not a bug.

**Comment by 4nanook** [comment]
When I go to the URL I see a preview so it would seem the data IS offered.  Firefox picks it up just fine.

**Comment by 4nanook** [comment]
> I don't think brandnewtube.com is general purpose enough that we should enable video embedding of their URLs in Friendica.

Youtube censors anything that conflicts with the mainstream narrative, alternatives like brandnewtube are needed for the same reason that alternatives like Friendica are needed to say Farcebook.

**Comment by MrPetovan** [comment]
And anyone is free not to accommodate Friendica in particular if we don't provide standard protocol endpoints.

**Comment by MrPetovan** [comment]
> When I go to the URL I see a preview so it would seem the data IS offered. Firefox picks it up just fine.

Where is this preview? Would you care to take a screenshot of it?

**Comment by 4nanook** [comment]
Here is an example, on Firefox it shows the video still just fine, but friendica does not scrape this and provide it when you put it in the video or URL link.

https://brandnewtube.com/watch/covid-19-vaccines-are-weapons-of-mass-destruction-and-could-wipe-out-the-human-race_GcjtJu9dY1RcSNh.html

**Comment by MrPetovan** [comment]
This is a link to the actual video page, where's the preview you are alluding to? Friendica uses a technical endpoint for embedding, and this endpoint isn't visible to humans even when it's implemented. Also given the title about egregious vaccine disinformation, I understand why they were rightfully banned from Youtube. Conflicting with mainstream media doesn't mean they're right.

**Comment by 4nanook** [comment]
The page shows a still of the video, friendica will not pick this up.

-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
  Eskimo North Linux Friendly Internet Access, Shell Accounts, and Hosting.
    Knowledgeable human assistance, not telephone trees or script readers.
  See our web site: http://www.eskimo.com/ (206) 812-0051 or (800) 246-6874.

**Comment by MrPetovan** [comment]
No, it's part of their web page, not of the technical endpoint I alluded to earlier which Friendica can read from. It's the reason why BitChute videos can be embedded but not BrandNewTube's. We didn't create an exception for BitChute, BrandNewTube just hasn't implemented the technical endpoint yet.

**Comment by 4nanook** [comment]
Ok well I guess I'll need to spend some time reading the documentation to understand this better.  I'm not much of a PHP guy, much more familiar with C or C++ but those don't lend themselves to web apps so well.

**Comment by MrPetovan** [comment]
So I ran a quick check to be sure about the technical side, and it turns out they do have the technical endpoint we can use to show a preview. What we don't have is the permission to access the page from Friendica nodes themselves. They must have some sort of User Agent filter in place since accessing the same URL you gave me from a Friendica script returns a 500 error while I was able to access it from my browser. I don't particularly blame them for not allowing unknown user agents but there's nothing more we can do on our end.
